lib/bot.rb

```diff
@@ -80,7 +81,7 @@ def pull_request_comment_content
 
   def extract_qa_comment
     search_pattern = /#{@magic_qa_keyword}.*\w+/i
-    return unless @comment.match?(search_pattern)
+    return unless @comment.match(search_pattern)
```

Comment:
Why do we use `match` instead of `match?`?

Reply:
```
bundle exec rspec
..FFFFFFFFFFFF.....

Failures:

  1) Bot#handle_comment when bot created comment does not create jira comment
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "I did it!":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:72:in `block (4 levels) in <top (required)>'

  2) Bot#handle_comment when linked issue exists when comment starts with QA: creates a jira comment with the rest of the comment and adds a reaction to the comment
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "QA: foo":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:87:in `block (5 levels) in <top (required)>'

  3) Bot#handle_comment when linked issue exists with multiline QA comments uses all lines
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "foo\nbar\nbaz QA: this\nis\na\nmultiline\ncomment":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:97:in `block (5 levels) in <top (required)>'

  4) Bot#handle_comment when linked issue exists when comment has QA: in the middle creates a jira comment with the rest of the comment
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "bar baz QA: foo":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:107:in `block (5 levels) in <top (required)>'

  5) Bot#handle_comment when linked issue exists with unrelated comment does not create a jira comment or a github reaction
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "Nice weather isn't it?":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:117:in `block (5 levels) in <top (required)>'

  6) Bot#handle_comment when linked issue exists with empty QA comment does not create a jira comment or a github reaction
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "QA:":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:127:in `block (5 levels) in <top (required)>'

  7) Bot#handle_comment when linked issue exists when the comment is edited does not create a jira comment or a github reaction
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "QA: foo":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:138:in `block (5 levels) in <top (required)>'

  8) Bot#handle_comment when linked issue exists with inline image parses image correctly
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for #<String:0x007f93f561fbf8>
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:148:in `block (5 levels) in <top (required)>'

  9) Bot#handle_comment when linked issue exists with headings parses headings correctly
     Failure/Error: return unless @comment.match?(search_pattern)

     NoMethodError:
       undefined method `match?' for "qa: # foo\n## bar\ncontent":String
       Did you mean?  match
     # ./lib/bot.rb:83:in `extract_qa_comment'
     # ./lib/bot.rb:29:in `handle_comment'
     # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
     # ./spec/bot_spec.rb:158:in `block (5 levels) in <top (required)>'

  10) Bot#handle_comment when linked issue doesn't exist creates the issue and renames the github PR
      Failure/Error: return unless @comment.match?(search_pattern)

      NoMethodError:
        undefined method `match?' for "QA: foo":String
        Did you mean?  match
      # ./lib/bot.rb:83:in `extract_qa_comment'
      # ./lib/bot.rb:29:in `handle_comment'
      # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
      # ./spec/bot_spec.rb:175:in `block (4 levels) in <top (required)>'

  11) Bot#handle_comment when linked issue doesn't exist when jira_transition_id is set creates the issue and renames the github PR
      Failure/Error: return unless @comment.match?(search_pattern)

      NoMethodError:
        undefined method `match?' for "QA: foo":String
        Did you mean?  match
      # ./lib/bot.rb:83:in `extract_qa_comment'
      # ./lib/bot.rb:29:in `handle_comment'
      # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
      # ./spec/bot_spec.rb:187:in `block (5 levels) in <top (required)>'

  12) Bot#handle_comment when linked issue doesn't exist with unrelated comment does not create a jira ticket, or comment or a github reaction
      Failure/Error: return unless @comment.match?(search_pattern)

      NoMethodError:
        undefined method `match?' for "Nice weather isn't it?":String
        Did you mean?  match
      # ./lib/bot.rb:83:in `extract_qa_comment'
      # ./lib/bot.rb:29:in `handle_comment'
      # ./spec/bot_spec.rb:57:in `block (3 levels) in <top (required)>'
      # ./spec/bot_spec.rb:199:in `block (5 levels) in <top (required)>'

Finished in 0.02567 seconds (files took 0.6336 seconds to load)
19 examples, 12 failures

Failed examples:

rspec ./spec/bot_spec.rb:70 # Bot#handle_comment when bot created comment does not create jira comment
rspec ./spec/bot_spec.rb:84 # Bot#handle_comment when linked issue exists when comment starts with QA: creates a jira comment with the rest of the comment and adds a reaction to the comment
rspec ./spec/bot_spec.rb:94 # Bot#handle_comment when linked issue exists with multiline QA comments uses all lines
rspec ./spec/bot_spec.rb:104 # Bot#handle_comment when linked issue exists when comment has QA: in the middle creates a jira comment with the rest of the comment
rspec ./spec/bot_spec.rb:114 # Bot#handle_comment when linked issue exists with unrelated comment does not create a jira comment or a github reaction
rspec ./spec/bot_spec.rb:124 # Bot#handle_comment when linked issue exists with empty QA comment does not create a jira comment or a github reaction
rspec ./spec/bot_spec.rb:135 # Bot#handle_comment when linked issue exists when the comment is edited does not create a jira comment or a github reaction
rspec ./spec/bot_spec.rb:145 # Bot#handle_comment when linked issue exists with inline image parses image correctly
rspec ./spec/bot_spec.rb:155 # Bot#handle_comment when linked issue exists with headings parses headings correctly
rspec ./spec/bot_spec.rb:170 # Bot#handle_comment when linked issue doesn't exist creates the issue and renames the github PR
rspec ./spec/bot_spec.rb:181 # Bot#handle_comment when linked issue doesn't exist when jira_transition_id is set creates the issue and renames the github PR
rspec ./spec/bot_spec.rb:194 # Bot#handle_comment when linked issue doesn't exist with unrelated comment does not create a jira ticket, or comment or a github reaction
```

How this ended up on master i don't know but half the tests were failing if it would keep the questionmark